common/djangoapps/util/tests/test_submit_feedback.py

```diff
@@ -303,7 +360,85 @@ def test_valid_request_auth_user(self, zendesk_mock_class, datadog_mock):
             )
         ]
         self.assertEqual(zendesk_mock_instance.mock_calls, expected_zendesk_calls)
-        self._assert_datadog_called(datadog_mock, with_tags=False)
+        self._assert_datadog_called(datadog_mock, [])
+
+    @data(
+        ("course-v1:testOrg+testCourseNumber+testCourseRun", True),
+        ("course-v1:testOrg+testCourseNumber+testCourseRun", False),
+        ("", None),
+        (None, None)
+    )
+    @unpack
+    @override_settings(ZENDESK_CUSTOM_FIELDS={"course_id": 1234, "enrollment_mode": 5678})
+    def test_valid_request_auth_user_with_custom_fields(self, course_id, enrollment_state, zendesk_mock, datadog_mock):
```

Comment:
This test has majority parts very similar to the `test_valid_request_anon_user_with_custom_fields` test case. Can you please refactor the common logic into a helper function to make it Dry?